### PR TITLE
Column Major Arrays

### PR DESCRIPTION
**Pull request by Dekken**
#260

### CODE REVIEW

**Comment by Dekken** [comment]
not working swig stuff on another branch https://github.com/Dekken/tick/tree/col_maj_swig

File: lib/swig/array/ssparsearray_typemap_in.i [hunk 1]
```diff
@@ -88,6 +87,10 @@
 // The final macro for dealing with sparse arrays
 %define SSPARSEARRAY_TYPEMAPIN_MACROS(SSPARSEARRAY_TYPE, SSPARSEARRAYPTR_TYPE, SSPARSEARRAY2D_TYPE, SSPARSEARRAY2DPTR_TYPE,
                                       SPARSEARRAY_TYPE, SPARSEARRAY2D_TYPE, C_TYPE)
+
 TYPEMAPIN_SSPARSEARRAY(SSPARSEARRAY_TYPE, SSPARSEARRAYPTR_TYPE, SPARSEARRAY_TYPE, SPARSEARRAY2D_TYPE, C_TYPE);
-TYPEMAPIN_SSPARSEARRAY2D(SSPARSEARRAY2D_TYPE, SSPARSEARRAY2DPTR_TYPE, SPARSEARRAY_TYPE, SPARSEARRAY2D_TYPE, C_TYPE)
+TYPEMAPIN_SSPARSEARRAY2D(SSPARSEARRAY2D_TYPE, SSPARSEARRAY2DPTR_TYPE, SPARSEARRAY_TYPE, SPARSEARRAY2D_TYPE, C_TYPE);
+//#error SColMaj##SPARSEARRAY2D_TYPE##
```
Comment (Mbompr) [inline lib/swig/array/ssparsearray_typemap_in.i]: This //error comment is frightening

File: tick/array/tests/serialize_test.py [hunk 2]
```diff
@@ -60,6 +60,22 @@ def test_serialize_sparse_2d_array(self):
...
+
+        serialized_array = load_array(self.array_file, array_dim=2,
+                                      array_type='sparse', dtype=self.dtype,
+                                      format="csc")
+        np.testing.assert_array_almost_equal(array.toarray(),
+                                             serialized_array.toarray())
+
+        # python 3.5 has show to required this - investigate typemappers
+        gc.collect()
+
```
Comment (Mbompr) [inline tick/array/tests/serialize_test.py]: Shouldn't we add a test for column major dense array?

File: appveyor.yml [hunk 3]
```diff
@@ -29,19 +29,10 @@ cache:
 
 install:
   - git submodule update --init
-  - curl -o mkn.exe -L https://ci.appveyor.com/api/projects/dekken/maiken/artifacts/bin/build/mkn.exe
   - IF NOT EXIST C:\ProgramData\chocolatey\bin\swig.exe choco install tools\windows\swig.3.0.12.nupkg --yes --limit-output #> $null
   - git clone https://github.com/X-DataInitiative/tick_appveyor -b master --depth 1 appveyor
 
 build_script:
-  - cp "C:\Program Files (x86)\Windows Kits\10\bin\x64\rc.exe" %CD%
```
[the remark this answers is not in the document]
Comment (Dekken) [reply appveyor.yml]: dropped the mkn build for gtests, times out - can put it in a different PR 
testing python does test C++

File: lib/swig/array/ssparsearray_typemap_in.i (hunk 1 above)
Comment (Dekken) [reply lib/swig/array/ssparsearray_typemap_in.i]: good catch - was testing if the file was even used

File: tick/array/tests/serialize_test.py (hunk 2 above)
Comment (Dekken) [reply tick/array/tests/serialize_test.py]: there's a gtest, we can move it to python

File: tick/array/tests/serialize_test.py [hunk 4]
```diff
@@ -60,6 +60,33 @@ def test_serialize_sparse_2d_array(self):
...
+    def test_serialize_column_major_2d_array(self):
+        """...Test serialization of 2d dense array is done as expected
+        """
+        x = np.array([[1, 2.1, 5], [0, 4, 1]]).astype(self.dtype)
+        array = np.asfortranarray(x).reshape(3, 2)
+        serialize_array(array, self.array_file)
+
+        serialized_array = load_array(self.array_file, array_dim=2,
+                                      dtype=self.dtype)
+        np.testing.assert_array_almost_equal(array, serialized_array)
```
Comment (Mbompr) [inline tick/array/tests/serialize_test.py]: What happens if you add
`np.testing.assert_array_almost_equal(x, serialized_array)` to compare it to its row major version ?

File: tick/array/tests/serialize_test.py [hunk 5]
```diff
@@ -60,6 +60,33 @@ def test_serialize_sparse_2d_array(self):
         # python 3.5 has show to required this - investigate typemappers
         gc.collect()
 
+    def test_serialize_column_major_2d_array(self):
+        """...Test serialization of 2d dense array is done as expected
+        """
+        x = np.array([[1, 2.1, 5], [0, 4, 1]]).astype(self.dtype)
+        array = np.asfortranarray(x).reshape(3, 2)
```
Comment (Mbompr) [inline tick/array/tests/serialize_test.py]: Don't reshape it, we want it to be the column major representation of a 2 lines, 3 columns array

File: lib/swig/array/array_test_cfunc.i [hunk 6]
```diff
@@ -18,10 +18,23 @@
 
         PyArrayObject *arrayObject = (PyArrayObject *) (obj);
 
-        if (!(PyArray_FLAGS(arrayObject) & NPY_ARRAY_C_CONTIGUOUS)) {
-            //PyErr_SetString(PyExc_ValueError,
-            //                "Numpy array data should be contiguous (use numpy.ascontiguousarray)");
-            //return(false);
+        if (std::is_same<ARRAY_TYPE::major_type, RowMajor>::value) {
```
[the remark this answers is not in the document]
Comment (Dekken) [reply lib/swig/array/array_test_cfunc.i]: dropped the check for column major - just if else now as it should only ever be row

**Comment by Dekken** [comment]
Failing due to timeouts (at least)

**Pull request merged**